sdk/servicebus/service-bus/test/atomXml.spec.ts

```diff
@@ -72,6 +76,59 @@ describe("atomSerializationPolicy #RunInBrowser", function() {
   });
 });
 
+describe("deserializeAtomXmlResponse #RunInBrowser", function() {
+  it("should throw an error if receiving a valid XML but invalid Atom XML", async function() {
+    try {
+      await serviceBusAtomManagementClient.deleteQueue("alwaysBeExistingQueue");
+    } catch (err) {
+      console.log("Ignoring clean up step");
+    }
+    const response = await serviceBusAtomManagementClient.createQueue("alwaysBeExistingQueue");
+
+    delete response._response["parsedBody"]["entry"];
+    delete response._response["parsedBody"]["feed"];
+    response._response["parsedBody"] = { notAnEntry: "" };
+
+    try {
+      await deserializeAtomXmlResponse(["QueueName"], response._response);
+      assert.deepEqual(true, false, "Error must be thrown");
+    } catch (err) {
+      assert.deepEqual(
+        err.message.startsWith(
+          "Error occurred while parsing the response body - expected the service to return atom xml content with either feed or entry elements."
+        ),
+        true,
+        `"${err.message}" was expected to begin with "Error occurred while parsing the response body - expected the service to return atom xml content with either feed or entry elements." `
+      );
+      assert.deepEqual(err.code, "PARSE_ERROR");
+    }
+  });
+
+  it("should throw appropriate error if unrecognized HTTP code is returned by service", async function() {
+    try {
+      await serviceBusAtomManagementClient.deleteQueue("alwaysBeExistingQueue");
+    } catch (err) {
+      console.log("Ignoring clean up step");
+    }
+    const response = await serviceBusAtomManagementClient.createQueue("alwaysBeExistingQueue");
+
+    response._response.status = 666;
```

Comment:
Same here, I don't think you actually have to make requests to the service if you create the `_response` yourself. There aren't really 'live' tests anyway.